src/Magento/FunctionalTestingFramework/Config/Reader/Filesystem.php

```diff
@@ -231,8 +231,13 @@ protected function validateSchema($configMerger, $filename = null)
         if ($this->validationState->isValidationRequired()) {
             $errors = [];
             if ($configMerger && !$configMerger->validate($this->schemaFile, $errors)) {
-                $message = $filename ? $filename . PHP_EOL . "Invalid Document \n" : PHP_EOL . "Invalid Document \n";
-                throw new \Exception($message . implode("\n", $errors));
+                foreach ($errors as $error){
+                    LoggingUtil::getInstance()->getLogger(Filesystem::class)->buildFailure(
+                        "XSD schema error ",
+                        [ "file"=> $filename ? $filename: ":mergedFile:", "error" => $error]
+                    );
+                }
+                throw new \Exception("Error: XSD schema issues found in file(s) " . $filename . "\n");
```

Comment:
I'm not sure I'm reading this right, but does this exception get thrown as soon as a single merged file fails the schema validation check? Or does `configMerger` iterate through every file?

Reply:
Not changing the current functionality; schema validation check fails at the first error encounter As discussed, will create a new story as this may not be a quick implementation.